storage_service/locations/models/package.py

```diff
@@ -1049,30 +1050,39 @@ def finish_reingest(self, origin_location, origin_path, reingest_location, reing
 
         # Compress
         # TODO update this for uncompressed AIPs
-        reingest_root = etree.parse(reingest_pointer)
-        os.remove(reingest_pointer)
-        puid = reingest_root.findtext('.//premis:formatRegistryKey', namespaces=utils.NSMAP)
-        if puid == 'fmt/484':  # 7 Zip
-            algo = reingest_root.find('.//mets:transformFile', namespaces=utils.NSMAP).get('TRANSFORMALGORITHM')
-            if algo == 'bzip2':
-                compression = self.COMPRESSION_7Z_BZIP
-            elif algo == 'lzma':
-                compression = self.COMPRESSION_7Z_LZMA
+        if self.is_compressed:
+            reingest_root = etree.parse(reingest_pointer)
+            os.remove(reingest_pointer)
+            puid = reingest_root.findtext('.//premis:formatRegistryKey', namespaces=utils.NSMAP)
+            if puid == 'fmt/484':  # 7 Zip
+                algo = reingest_root.find('.//mets:transformFile', namespaces=utils.NSMAP).get('TRANSFORMALGORITHM')
+                if algo == 'bzip2':
+                    compression = self.COMPRESSION_7Z_BZIP
+                elif algo == 'lzma':
+                    compression = self.COMPRESSION_7Z_LZMA
+                else:
+                    compression = self.COMPRESSION_7Z_BZIP
+                    LOGGER.warning('Reingest: Unable to determine reingested compression algorithm, defaulting to bzip2.')
+            elif puid == 'x-fmt/268':  # Bzipped (probably tar)
+                compression = self.COMPRESSION_TAR_BZIP2
             else:
                 compression = self.COMPRESSION_7Z_BZIP
-                LOGGER.warning('Reingest: Unable to determine reingested compression algorithm, defaulting to bzip2.')
-        elif puid == 'x-fmt/268':  # Bzipped (probably tar)
-            compression = self.COMPRESSION_TAR_BZIP2
-        else:
-            compression = self.COMPRESSION_7Z_BZIP
-            LOGGER.warning('Reingest: Unable to determine reingested file format, defaulting recompression algorithm to %s.', compression)
-        LOGGER.info('Reingest: compressing with %s', compression)
-        # FIXME Do we need compression output for event?
-        out_path, out_dir = self.compress_package(compression)
-        compress_output = ''
+                LOGGER.warning('Reingest: Unable to determine reingested file format, defaulting recompression algorithm to %s.', compression)
+            LOGGER.info('Reingest: compressing with %s', compression)
+            # FIXME Do we need compression output for event?
+            out_path, out_dir = self.compress_package(compression)
+            compress_output = ''
 
         # Delete working files
         shutil.rmtree(reingest_full_path)
+
+        if not self.is_compressed:
+            # If temp files are deleted before compression check, the wrong result is returned 
+            shutil.rmtree(temp_dir)
+            self.save()
+            return
```

Comment:
This skips all the code that re-stores the updated AIP.  Was this supposed to be at the end of this function?